chain/chain/src/chain.rs

```diff
@@ -2415,6 +2466,34 @@ impl Chain {
             return Err(Error::InvalidBlockHeight(block.height()));
         }
 
+        // A heuristic to check signature of the block even if prev block hash is not saved yet
+        let epoch_ids =
+            match self.epoch_manager.get_epoch_id_from_prev_block(&block.prev_block_hash()) {
```

Comment:
This function does things very similar to `validate_chunk_relevant` in chain/client/src/stateless_validation/validate.rs

The idea of chain/client/src/stateless_validation/validate.rs was so that we could have all the common validation code like this in one place and make it easy to identify vulnerabilities.

If we are in a time crunch, I guess it's okay to have this, but might I suggest trying to consolidate any validation code in chain/client/src/stateless_validation/validate.rs?

Reply:
Yeah, definitely worth following up when pressure is lesser or else we'll forget easily about this..